### ISSUE TITLE
Race Condition

### ISSUE THREAD

**Issue by cmueller-tp** [Velocidex/velociraptor#2079]
I've just had this race condition on the server side with one connection to the frontend and ~70 clients:
```
==================
WARNING: DATA RACE
Read at 0x00c0016fd420 by goroutine 560:
  www.velocidex.com/golang/velociraptor/services/hunt_dispatcher.(*HuntDispatcher).ListHunts()
      <autogenerated>:1 +0x84
  www.velocidex.com/golang/velociraptor/api.(*ApiServer).ListHunts()
      /root/velociraptor/api/hunts.go:206 +0x1bd
  www.velocidex.com/golang/velociraptor/api/proto._API_ListHunts_Handler()
      /root/velociraptor/api/proto/api_grpc.pb.go:1057 +0x25d
  google.golang.org/grpc.(*Server).processUnaryRPC()
      /root/go/pkg/mod/google.golang.org/grpc@v1.42.0/server.go:1282 +0x13ea
  google.golang.org/grpc.(*Server).handleStream()
      /root/go/pkg/mod/google.golang.org/grpc@v1.42.0/server.go:1616 +0xff8
  google.golang.org/grpc.(*Server).serveStreams.func1.2()
      /root/go/pkg/mod/google.golang.org/grpc@v1.42.0/server.go:921 +0xec

Previous write at 0x00c0016fd420 by goroutine 559:
  sync/atomic.AddInt32()
      /usr/local/go/src/runtime/race_amd64.s:281 +0xb
  sync/atomic.AddInt32()
      <autogenerated>:1 +0x1a
  www.velocidex.com/golang/velociraptor/services/hunt_dispatcher.(*HuntDispatcher).GetHunt()
      /root/velociraptor/services/hunt_dispatcher/hunt_dispatcher.go:250 +0xf7
  www.velocidex.com/golang/velociraptor/api.(*ApiServer).GetHunt()
      /root/velociraptor/api/hunts.go:262 +0x2d0
  www.velocidex.com/golang/velociraptor/api/proto._API_GetHunt_Handler()
      /root/velociraptor/api/proto/api_grpc.pb.go:1075 +0x25d
  google.golang.org/grpc.(*Server).processUnaryRPC()
      /root/go/pkg/mod/google.golang.org/grpc@v1.42.0/server.go:1282 +0x13ea
  google.golang.org/grpc.(*Server).handleStream()
      /root/go/pkg/mod/google.golang.org/grpc@v1.42.0/server.go:1616 +0xff8
  google.golang.org/grpc.(*Server).serveStreams.func1.2()
      /root/go/pkg/mod/google.golang.org/grpc@v1.42.0/server.go:921 +0xec
      Goroutine 560 (running) created at:
  google.golang.org/grpc.(*Server).serveStreams.func1()
      /root/go/pkg/mod/google.golang.org/grpc@v1.42.0/server.go:919 +0x4dd
  google.golang.org/grpc/internal/transport.(*http2Server).operateHeaders()
      /root/go/pkg/mod/google.golang.org/grpc@v1.42.0/internal/transport/http2_server.go:588 +0x30be
  google.golang.org/grpc/internal/transport.(*http2Server).HandleStreams()
      /root/go/pkg/mod/google.golang.org/grpc@v1.42.0/internal/transport/http2_server.go:633 +0x378
  google.golang.org/grpc.(*Server).serveStreams()
      /root/go/pkg/mod/google.golang.org/grpc@v1.42.0/server.go:905 +0x275
  google.golang.org/grpc.(*Server).handleRawConn.func1()
      /root/go/pkg/mod/google.golang.org/grpc@v1.42.0/server.go:847 +0x64

Goroutine 559 (running) created at:
  google.golang.org/grpc.(*Server).serveStreams.func1()
      /root/go/pkg/mod/google.golang.org/grpc@v1.42.0/server.go:919 +0x4dd
  google.golang.org/grpc/internal/transport.(*http2Server).operateHeaders()
      /root/go/pkg/mod/google.golang.org/grpc@v1.42.0/internal/transport/http2_server.go:588 +0x30be
  google.golang.org/grpc/internal/transport.(*http2Server).HandleStreams()
      /root/go/pkg/mod/google.golang.org/grpc@v1.42.0/internal/transport/http2_server.go:633 +0x378
  google.golang.org/grpc.(*Server).serveStreams()
      /root/go/pkg/mod/google.golang.org/grpc@v1.42.0/server.go:905 +0x275
  google.golang.org/grpc.(*Server).handleRawConn.func1()
      /root/go/pkg/mod/google.golang.org/grpc@v1.42.0/server.go:847 +0x64
==================

**Comment by scudette** [comment]
Thanks for reporting it - normally building with the race detector will produce a very slow binary but it is good to test :-) Will send a fix soon

**Comment by scudette** [comment]
Should be fixed by #2081
